include/nighthawk/common/poolable.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+
+#include "envoy/common/pure.h"
+
+#include "common/common/non_copyable.h"
+
+namespace Nighthawk {
+
+class Poolable : Envoy::NonCopyable {
```

Comment:
Can you write some comments here about the intended usage? For example, what happens when a `Poolable` is orphaned? Who is responsible for deleting it?

Reply:
done, hope these comments are clarifying

Reply:
Yup, thanks! So, if my understanding is correct, I think `Poolable` isn't the right design for this. That is, I don't think you should need to implement `Poolable` for a class that you want to be manageable by a pool. As an analogy, you don't need to implement a `Shareable` interface to be managed by a `shared_ptr`.

Instead, I think you probably want to create a `PoolPtr<T>` class to capture the semantics you intend. It will probably behave a bit like a `unique_ptr` (in the sense that only one owner can borrow from the pool at a time, so it can only be moved and not copied), but might also have a `bool is_orphaned_` that the pool can set directly. If you can implement it just using `unique_ptr` with a custom deleter, the way you're already mostly doing, even better...

But, yeah, I agree with Josh and Harvey's comments: we should probably get some good evidence to support the need for this first.

Reply:
Yes I agree the high level question should be addressed first. That aside, I do really like the idea you describe here, I feel following up on that would yield a huge conceptual improvement to this